src/main/java/com/nashtech/service/impl/FirestoreDbService.java

```diff
@@ -0,0 +1,98 @@
+package com.nashtech.service.impl;
+
+import com.google.cloud.spring.data.firestore.FirestoreDataException;
+import com.nashtech.exception.DataNotFoundException;
+import com.nashtech.model.Car;
+import com.nashtech.model.CarBrand;
+import com.nashtech.repository.FirestoreDbRepository;
+import com.nashtech.service.CloudDataService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+
+import java.time.Duration;
+
+/**
+ * FirestoreDbService is an implementation
+ * of the CloudDataService interface that interacts
+ * with Firestore database to retrieve Car and CarBrand data.
+ */
+@Slf4j
+@Service
+@Profile("firestore")
+public class FirestoreDbService implements CloudDataService {
+
+    /**
+     * The duration of the interval,
+     * in milliseconds, for retrieving brand and car details.
+     */
+    private static final Integer DELAY_TIME = 100;
+    /**
+     * The VehicleRepository instance used to retrieve car information.
+     */
+    @Autowired
+    private FirestoreDbRepository firestoreDbRepository;
+
+    /**
+     * Retrieves all CarBrands from Firestore database.
+     * @return A Flux of CarBrand objects.
+     *
+     */
+    @Override
+    public Flux<CarBrand> getAllBrands() {
+        return firestoreDbRepository.findAll()
+                .filter(gcpCarEntity -> gcpCarEntity.getBrand() != null)
+                .map(gcpCarEntity -> new CarBrand(gcpCarEntity.getBrand()))
+                .distinct()
+                .delayElements(Duration.ofMillis(DELAY_TIME))
+                .doOnNext(carBrand -> log.info("Brand: " + carBrand))
```

Comment:
Not needed. Can work with doOnComplete only

Reply:
okay updated the  code